lib/qs/dispatch_job_handler.rb

```diff
@@ -0,0 +1,69 @@
+require 'qs'
+require 'qs/dispatch_job'
+require 'qs/job_handler'
+
+module Qs
+
+  module DispatchJobHandler
+
+    def self.included(klass)
+      klass.class_eval do
+        include Qs::JobHandler
+        include InstanceMethods
+      end
+    end
+
+    module InstanceMethods
+
+      attr_reader :event, :subscribed_queue_names
+
+      def init!
+        @event = Qs::DispatchJob.event(job)
+        @subscribed_queue_names = Qs.event_subscribers(@event)
+        @qs_failed_dispatches = []
+      end
+
+      def run!
+        logger.info "Dispatching #{self.event.route_name}"
+        logger.info "  params:       #{self.event.params.inspect}"
+        logger.info "  publisher:    #{self.event.publisher}"
+        logger.info "  published at: #{self.event.published_at}"
+        logger.info "Found #{self.subscribed_queue_names.size} subscribed queue(s)"
+        self.subscribed_queue_names.each do |queue_name|
+          qs_dispatch(queue_name, self.event)
+        end
+        qs_handle_errors(self.event, @qs_failed_dispatches)
+      end
+
+      private
+
+      def qs_dispatch(queue_name, event)
+        Qs.push(queue_name, Qs::Payload.event_hash(event))
+        logger.info "  => #{queue_name}"
+      rescue StandardError => exception
+        logger.info "  => #{queue_name} (failed)"
+        @qs_failed_dispatches << FailedDispatch.new(queue_name, exception)
+      end
+
+      def qs_handle_errors(event, failed_dispatches)
+        return if failed_dispatches.empty?
+        logger.info "Failed to dispatch the event to " \
+                    "#{failed_dispatches.size} subscribed queues"
+        queue_names = failed_dispatches.map do |fail|
+          logger.info "#{fail.queue_name}"
+          logger.info "  #{fail.exception.class}: #{fail.exception.message}"
+          logger.info "  #{fail.exception.backtrace.first}"
+          fail.queue_name
+        end
+        raise DispatchError, "#{event.route_name} event wasn't " \
+                             "dispatched to: #{queue_names.join(', ')}"
```

Comment:
@jcredding do you think we should add the failed exception class and message to the dispatch error so the error handler can give at least a clue to the developers?  I guess maybe we'd just have to choose the first fail or something?  Just a thought.